app/react/Map/Map.js

```diff
@@ -61,7 +61,7 @@ export default class Map extends Component {
     const longitude = props.longitude || viewport.longitude;
     const newViewport = Object.assign(viewport, { latitude, longitude, markers });
 
-    if (JSON.stringify(props.markers) !== JSON.stringify(markers)) {
+    if (JSON.stringify(this.props.markers) !== JSON.stringify(markers)) {
```

Comment:
JSON.stringify is not the most reliable way to compare objects, [check](https://stackoverflow.com/questions/15376185/is-it-fine-to-use-json-stringify-for-deep-comparisons-and-cloning)  first and second answer